CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CIReinforcementsManager.uc

```diff
@@ -0,0 +1,122 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Manager class to control the delay of any reinforcements issued
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_CIReinforcementsManager extends XComGameState_BaseObject;
+
+var array<DelayedReinforcementSpawner> DelayedReinforcementSpawners;
+
+var int NextReinforcementsArrival;
+var int LastTurnModified;
+
+var localized string m_strReinforcementsBodyWarning;
+var localized string m_strReinforcementsBodyImminent;
+
+static function CreateReinforcementsManager()
+{
+	local XComGameState NewGameState;
+
+	if(GetReinforcementsManager(true) == none)
+	{
+		NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Creating reinforcements manager singleton");
+
+		NewGameState.CreateNewStateObject(class'XComGameState_CIReinforcementsManager');
+
+		`TACTICALRULES.SubmitGameState(NewGameState);
+	}
+}
+
+static function XComGameState_CIReinforcementsManager GetReinforcementsManager(optional bool AllowNull = false)
+{
+	return XComGameState_CIReinforcementsManager(`XCOMHISTORY.GetSingleGameStateObjectForClass(class'XComGameState_CIReinforcementsManager', AllowNull));
+}
+
+function AddDelayedReinforcementSpawner(DelayedReinforcementSpawner DelayedSpawner, optional XComGameState_CIReinforcementsManager ManagerState, optional bool bResetUpdate=false)
+{
+	local XComGameState NewGameState;
+
+	if (ManagerState == none)
+	{
+		NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Deleting DelayedReinforcementSpawner from manager");
```

Comment:
Don't do this here. Just assume whatever calls `AddDelayedReinforcementSpawner` handle gamestate modification and such